### PR TITLE
Amount datatype needs to be cast to ensure proper output

### PR DESCRIPTION
**Pull request by mattprintz**
Forces schema in resulting dataframe to have the proper column type, even if it is filled with NULL or empty.

### CODE REVIEW

File: spark_auto_mapper/data_types/amount.py [hunk 1]
```diff
@@ -23,15 +23,15 @@ def __init__(self, value: AutoMapperAmountInputType):
     def get_column_spec(
         self, source_df: Optional[DataFrame], current_column: Optional[Column]
     ) -> Column:
-        if source_df is not None and isinstance(self.value, AutoMapperDataTypeColumn) \
-                and dict(source_df.dtypes)[self.value.value] not in ("float", "double"):
-            # parse the amount here
+        if isinstance(self.value, AutoMapperDataTypeColumn) \
```
Comment (adhearn) [inline spark_auto_mapper/data_types/amount.py:26]: Is it clearer to write this like:

```
column_spec = self.value.get_column_spec(
    source_df=source_df, current_column=current_column
)
if <all the if stuff>:
    column_spec = column_spec.cast("double")

return column_spec
```

I think it's equivalent, and is a bit more obvious about what's happening here.

**Review by adhearn — COMMENTED** [review]
It looks correct, but I think we might be able to make the function a bit more readable, let me know what you think.

File: spark_auto_mapper/data_types/amount.py (hunk 1 above)
Comment (mattprintz) [reply spark_auto_mapper/data_types/amount.py]: Updated.

**Review by amin010 — APPROVED** [review]
LGTM!

**Review by imranq2 — APPROVED** [review]

**Pull request merged**